src/EditorFeatures/Core/Host/IStreamingFindReferencesPresenter.cs

```diff
@@ -100,11 +103,15 @@ internal static class IStreamingFindUsagesPresenterExtensions
                 // There was only one location to navigate to.  Just directly go to that location. If we're directly
                 // going to a location we need to activate the preview so that focus follows to the new cursor position.
 
-                return nonExternalItems[0].TryNavigateTo(workspace, showInPreviewTab: true, activateTab: true, cancellationToken);
+                return await nonExternalItems[0].TryNavigateToAsync(
+                    workspace, showInPreviewTab: true, activateTab: true, cancellationToken).ConfigureAwait(false);
             }
 
             if (presenter != null)
             {
+                // Can only navigate or present items on UI thread.
+                await threadingContext.JoinableTaskFactory.SwitchToMainThreadAsync(cancellationToken);
```

Comment:
@CyrusNajmabadi there's a bunch of ConfigureAwait(false)es after this -- expected?

Reply:
taken offline.  yes :)